server/server.go

```diff
@@ -538,24 +588,68 @@ func (server HTTPServer) findRule(
 	return rule, true
 }
 
-func (server HTTPServer) singleRuleEndpoint(writer http.ResponseWriter, request *http.Request) {
-	ruleID, err := readRuleID(writer, request)
+func (server HTTPServer) fetchAggregatorReportRule(
+	writer http.ResponseWriter, request *http.Request,
+) (*types.RuleOnReport, bool) {
+	clusterID, successful := httputils.ReadClusterName(writer, request)
+	// Error message handled by function
+	if !successful {
+		return nil, false
+	}
+
+	ruleID, errorKey, err := readRuleIDWithErrorKey(writer, request)
 	if err != nil {
-		return
+		return nil, false
 	}
 
-	aggregatorResponse, successful := server.fetchAggregatorReport(writer, request)
-	// Error message handled by function
+	authToken, err := server.GetAuthToken(request)
+	if err != nil {
+		handleServerError(writer, err)
+		return nil, false
+	}
+
+	userID := authToken.AccountNumber
+	orgID := authToken.Internal.OrgID
+
+	aggregatorResponse, successful := server.readAggregatorRuleForClusterID(orgID, clusterID, userID, ruleID, errorKey, writer)
 	if !successful {
-		return
+		return nil, false
 	}
+	return aggregatorResponse, true
+}
 
-	rule, successful := server.findRule(writer, aggregatorResponse.Report, ruleID)
+func (server HTTPServer) singleRuleEndpoint(writer http.ResponseWriter, request *http.Request) {
+	var rule proxy_types.RuleWithContentResponse
+
+	aggregatorResponse, successful := server.fetchAggregatorReportRule(writer, request)
 	// Error message handled by function
 	if !successful {
 		return
 	}
 
+	ruleWithContent, err := content.GetRuleWithErrorKeyContent(aggregatorResponse.Module, aggregatorResponse.ErrorKey)
+	if err != nil {
+		handleServerError(writer, err)
+		return
+	}
+
+	rule = proxy_types.RuleWithContentResponse{
+		CreatedAt:    ruleWithContent.PublishDate.UTC().Format(time.RFC3339),
+		Description:  ruleWithContent.Description,
+		ErrorKey:     aggregatorResponse.ErrorKey,
+		Generic:      ruleWithContent.Generic,
+		Reason:       ruleWithContent.Reason,
+		Resolution:   ruleWithContent.Resolution,
+		TotalRisk:    ruleWithContent.TotalRisk,
+		RiskOfChange: ruleWithContent.RiskOfChange,
+		RuleID:       aggregatorResponse.Module,
+		TemplateData: aggregatorResponse.TemplateData,
+		Tags:         ruleWithContent.Tags,
+		UserVote:     aggregatorResponse.UserVote,
+		Disabled:     aggregatorResponse.Disabled,
```

Comment:
looks like it lacks DisabledFeedback and DisabledAt attributes https://github.com/RedHatInsights/insights-results-smart-proxy/blob/master/server/server.go#L522 Even better would be to make some converting function probably in proxy_types package so that we have only one place to edit